docs/source/framework/pytorch_integration/layers_database.rst

```diff
@@ -0,0 +1,364 @@
+    }
+
+Concat
+^^^^^^
+
+.. code::
+
+    def concat(float(M, N) A, float(M, N) B) -> (O1) {
+        O1(n, i, m) = i == 0 ? A(m, n) : B(m, n) where i in 0:2
+    }
+
+Cast
+^^^^
+
+.. code::
+
+    def cast(float(M,N) A) -> (int32(M,N) O1) {{
+        O1(m, n) = int32(A(m, n) + {four})
+    }}
+
+Copy
+^^^^
+
+.. code::
+
+    def copy(float(M, N) I) -> (O) {
+        O(i, j) = I(i, j)
+    }
+
+Scale
+^^^^^
+
+.. code::
+
+    def scale(float(M, N) I) -> (O) {{
+        O(m, n) = I(m, n) * {s}
+    }}
+
+Fused Layers
+------------
+
+FCRelu
+^^^^^^
+
+.. code::
+
+    def fcrelu(float(B,M) I, float(N,M) W1, float(N) B1) -> (O1){
+        O1(b, n) +=! I(b, m) * W1(n, m)
+        O1(b, n) = O1(b, n) + B1(n)
+        O1(b, n) = fmax(O1(b, n), 0)
+    }
+
+Small MobileNet
+^^^^^^^^^^^^^^^
+
+.. code::
+
+    def small_mobilenet(float(C1, H, W) I, float(C1, KH1, KW1) W1, float(C1) B1, float(C2, C1) W2, float(C2) B2)
+    -> (O1, O2) {
+        O1(c1, h, w) +=! I(c1, h + kh, w + kw) * W1(c1, kh, kw)
+        O1(c1, h, w)  = O1(c1, h, w) + B1(c1)
+        O1(c1, h, w)  = fmax(O1(c1, h, w), 0)
+
+        O2(c2, h, w) +=! O1(c1, h, w) * W2(c2, c1)
+        O2(c2, h, w)  = O2(c2, h, w) + B2(c2)
+        O2(c2, h, w)  = fmax(O2(c2, h, w), 0)
+    }
+
+Normalization Layers
+--------------------
+
+Batch Normalization
+^^^^^^^^^^^^^^^^^^^
+
+.. code::
+
+    def batchnorm(float(N,C,H,W) I, float(C) rMeanIn, float(C) rVarIn)
+    -> (O, rMeanOut, rVarOut, mean, centered, variance, expectedVariance, normalizedOut)
+    {{
+        mean(c) +=! I(nn, c, hh, ww)
+        mean(c)  = mean(c) / (N * H * W)
+        rMeanOut(c) = (1 - {momentum}) * rMeanIn(c) + {momentum} * mean(c)
+        centered(n, c, h, w) = I(n, c, h, w) - rMeanOut(c)
+        variance(n, c, h, w) = centered(n, c, h, w) * centered(n, c, h, w)
+        expectedVariance(c) +=! (variance(n, c, h, w) + {eps}) / (N * H * W)
+        rVarOut(c) = rsqrt((1 - {momentum}) * rVarIn(c) + {momentum} * expectedVariance(c))
+        O(n, c, h, w) = centered(n, c, h, w) * rVarOut(c)
+        normalizedOut(n, c, h, w) = O(n, c, h, w)
+    }}
+
+Layer Normalization
+^^^^^^^^^^^^^^^^^^^
+
+.. code::
+
+    def layernorm(float(T, B, C) I) -> (O, mean, centered, var) {{
+        mean(t, b) +=! I(t, b, c) / C
+        centered(t, b, c) = I(t, b, c) - mean(t, b)
+        var(t, b) +=! centered(t, b, c) * centered(t, b, c)
+        var(t, b) = (var(t, b) + {eps}) / C
+        O(t, b, c) = centered(t, b, c) / rsqrt(var(t, b))
+    }}
+
+Distance Functions
+------------------
+
+Cosine Similarity
+^^^^^^^^^^^^^^^^^
+
+.. code::
+
+    def cosine_similarity(float(M, N) I1, float(M, N) I2) -> (O, sumI1, sumI2) {{
+        sumI1(m) +=! I1(m, n) * I1(m, n)
+        sumI2(m) +=! I2(m, n) * I2(m, n)
+        O(m) +=! (I1(m, n) * I2(m, n)) / fmax(rsqrt(sumI1(m)) * sqrt(sumI2(m)), {eps})
+    }}
+
+What operations can not be expressed
+------------------------------------
+* **Reshaping** Tensors inside the language
```

Comment:
For reshaping there are most often workarounds  in pure TC that make the intent of the program much better, examples can be added in the future as I think this file should go for the reason exposed above.

Reply:
IMO we should keep this file. As a user myself, I find it very convenient to be able to look at all the layers together. Rather than removing file, we need to do something better. THe issue I opened around creating a library for TC defs will help bringing structure to how we use and define and where we keep the TC defs. But that will happen later after the release.